### PR TITLE
Homework Podstawy C++ #2

### PR DESCRIPTION
**Pull request by MarekJanek**
Authors: @LogarithmDerivative, @MarekJanek

### CODE REVIEW

File: module2/homework/generateSequence/generateSequence.hpp [hunk 1]
```diff
@@ -1,7 +1,19 @@
...
+std::vector<int> generateSequence(int count, int step)
+{
+    if(count > 0)
+    {
+        std::vector<int> result;
+        int value = step;
+
+        for(size_t i = 0; i < count; i++)
+        {
+            result.push_back(value);
```
Comment (piezol) [inline module2/homework/generateSequence/generateSequence.hpp:13]: Memory will have to be allocated several times.
The size of result vector is known beforehend (== count), so we should allocate memory once using reserve

File: module2/homework/maxOfVector/maxOfVector.hpp [hunk 2]
```diff
@@ -3,6 +3,13 @@
...
-    return {};
+    
+    int max=std::numeric_limits<int>::min();
+    
+    for(const int& element : vec)
+    {
+        if(element>max)
+            max=element;
+    }
+    return max;
```
Comment (piezol) [inline module2/homework/maxOfVector/maxOfVector.hpp:14]: That has defect of returning std::numberic_limits<int>::min() for empty vector.
But the behaviour was unspecified for that case, so i guess anything works.

File: module2/homework/nwdNww/nwdNww.hpp [hunk 3]
```diff
@@ -1,11 +1,36 @@
...
-    // TODO: Implement me :)
-    return -1;
+
+    if(lhs==0 && rhs==0)
+        return 0;
+    
+    lhs=abs(lhs);
+    rhs=abs(rhs); 
+
+    return (lhs*rhs)/NWD(lhs,rhs);
```
Comment (piezol) [inline module2/homework/nwdNww/nwdNww.hpp:34]: There are trivial cases that your code won't work for, ie NWW(1e8, 1e8).
That is due to lhs*rhs overflowing int.
You can fix it by casting lhs to unsigned long long in the multiplication.

There are some nontrivial cases that will still fail, ie NWW(1e8-1, 1e8+1), but they fail for the std:lcm(int, int) as well, so it's ok.

File: module2/homework/generateSequence/generateSequence.hpp (hunk 1 above)
Comment (Rosein) [reply module2/homework/generateSequence/generateSequence.hpp:13]: Or instead using reserve you can set well size of vector at its creation.

File: module2/homework/maxOfVector/maxOfVector.hpp [hunk 4]
```diff
@@ -3,6 +3,13 @@
 #include <limits>
 
 int maxOfVector(const std::vector<int>& vec) {
-    // TODO: Implement me :)
-    return {};
+    
+    int max=std::numeric_limits<int>::min();
```
Comment (nauka-programowania-MA) [inline module2/homework/maxOfVector/maxOfVector.hpp:7]: beter use max = vec[0]

**Comment by ziobron** [comment]
{ "requirement_ids": [1119, 1631, 1119, 1633, 1119, 1634, 1119, 1635], "points": [2, 4, 2, 6, 2, 5, 2, 5] }

**Pull request closed without merging**